### ISSUE TITLE
Font File not found Taichi 1.0.2

### ISSUE THREAD

**Issue by joshDavy1** [taichi-dev/taichi#5051]
**Describe the bug**
Any script which attempts to show gui text will crash with the following error stating the Font file is not found.

**To Reproduce**

```py
import taichi as ti
from time import sleep
ti.init(ti.cpu)

gui = ti.GUI("Test", (100, 100))
while gui.running:
    gui.text("Test", pos=(0.5, 0.5))
    gui.show()
    sleep(0.1)
```

**Log/Screenshots**
Please post the **full log** of the program (instead of just a few lines around the error message, unless the log is > 1000 lines). This will help us diagnose what's happening. For example:

```
python test.py                                            
[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, win, python 3.8.13                                                                   
[Taichi] Starting on arch=x64                                                                                                              
[E 05/25/22 16:29:04.188 14636] [image_buffer.cpp:taichi::ArrayND<2,struct taichi::VectorND<4,float,0> >::write_text@120]
 Font file not found: C:\ProgramData\Anaconda3\envs\py38\Lib\site-packages\taichi\_lib\runtime/../../assets/Go-Regular.ttf
                                                                                            
File "test.py", line 7, in <module>                                                                                                          
gui.text("Test", pos=(0.5, 0.5))                                                                                                         
File "C:\ProgramData\Anaconda3\envs\py38\lib\site-packages\taichi\ui\gui.py", line 652, in text
self.canvas.text(content, pos, font_size, color) 
RuntimeError: [image_buffer.cpp:taichi::ArrayND<2,struct taichi::VectorND<4,float,0> >::write_text@120] 
Font file not found: C:\ProgramData\Anaconda3\envs\py38\Lib\site-packages\taichi\_lib\runtime/../../assets/Go-Regular.ttf   
```

**Comment by joshDavy1** [comment]
To add, I have tried reinstalling and also the nightly build and have reached the same bug.

**Comment by qiao-bo** [comment]
Thanks for the info. This probably relates to the recent change on `runtime_lib_dir`. We will take a look.

**Comment by qiao-bo** [comment]
Can also be reproduced on MacOS.

**Comment by qiao-bo** [comment]
Update: After some investigation, it seems that our new skbuild based setup function works differently than the old setuptools implementation (what we use prior to v1.0.1). Specifically. we need to provide the `external/assets` as a source in our package. IIUC, this should be executed like: First, the `assets` folder is copied from external to taichi source tree by this:
https://github.com/taichi-dev/taichi/blob/2197fad07a03f2142781a7e308c7eb9ed001cd5c/setup.py#L77
Then, the folder is again copied to the package by locating in `MANIFEST.in` 
https://github.com/taichi-dev/taichi/blob/2197fad07a03f2142781a7e308c7eb9ed001cd5c/MANIFEST.in#L2
Finally, by pip installing the wheel, we should see the `assets` folder in the desired location, such as `site-packages\taichi\_lib\runtime/../../assets`.

Now, the order is incorrect in my local log output. First, skbuild tempts to copy packages in `MANIFEST.in`. It cannot find the assets folder in the source tree. Then, at some point, `egg_info` will be executed to copy the assets folder from external to in-tree source. As a result, the `assets` folder is never packaged. (PS, i expect ti gallery does not work as well...)

I can see two solutions: 1) by always executing `python setup.py egg_info` before `bdist_wheel`. Thus i can force `egg_info` to run before the `MANIFEST.in` step. or 2) add a command in CMake to copy the `assets` folder, since in the end we just need the file in the source tree. @frostming @ailzhang WDYT?

**Comment by qiao-bo** [comment]
Update: Should be fixed by #5063.
